packages/flask/src/UIAnimatedBalance/UIAnimatedBalance.tsx

```diff
@@ -0,0 +1,286 @@
+     * Example:
+     * 1234.67 -> 5555.55
+     *
+     * We render icon with hidden part, as it changes width
+     * less frequently then visible one, therefore it looks stable.
+     * Visible part is just placed above with absolute positioning.
+     */
+    const normalized = useSharedValue(value.toString());
+    useAnimatedReaction(
+        () => {
+            return formatted.value;
+        },
+        (state, prevState) => {
+            // To change the hidden input only in case the length is changed
+            if (
+                prevState != null &&
+                state.integer.length + state.fractional.length ===
+                    prevState.integer.length + prevState.fractional.length
+            ) {
+                return;
+            }
+
+            const numRegExp = /[0-9]/g;
+            const wideNumLiteral = '5';
+
+            normalized.value =
+                state.integer.replace(numRegExp, wideNumLiteral) +
+                decimalSeparator +
+                state.fractional.replace(numRegExp, wideNumLiteral);
+        },
+    );
+
+    /**
+     * Basically we exploit the fact that TextView's text
+     * can be updated with `setNativeProps`,
+     * in order to not do it manually just delegating it here
+     * to reanimated, that will do the same under the hood
+     */
+    const animatedIntegerProps: any = useAnimatedProps(() => {
+        return {
+            text: formatted.value.integer,
+        };
+    });
+    const animatedFractionalProps: any = useAnimatedProps(() => {
+        return {
+            text: formatted.value.fractional,
+        };
+    });
+
+    const normalizedProps: any = useAnimatedProps(() => {
+        return {
+            text: normalized.value,
+        };
+    });
+
+    const iconElement = useIcon(icon, integerVariant, loading);
+
+    const colorStyle = React.useMemo(() => ({ color: theme[ColorVariants.TextPrimary] }), [theme]);
+
+    const textLikeContainer = React.useMemo(
+        () => (I18nManager.isRTL ? styles.rtlContainer : styles.ltrContainer),
+        [I18nManager.isRTL],
+    );
+
+    return (
+        <View style={styles.intrinsicWrapper}>
+            <View style={[textLikeContainer, styles.main]}>
+                <AnimatedTextInput
+                    style={[Typography[integerVariant], styles.hidden]}
+                    animatedProps={normalizedProps}
+                    defaultValue={normalized.value}
+                    underlineColorAndroid="transparent"
+                    editable={false}
+                />
+                {/*
+                 * Space letter here is important here, it's not a mistake!
+                 * It's placed to have a separation from main balance,
+                 * instead of `marginLeft` as space is more "text friendly"
+                 * and will layout properly in text
+                 */}
+                <Text style={Typography[fractionalVariant]}> {iconElement}</Text>
```

Comment:
Will this white space work fine with RTL?

Reply:
Yep, text is rendered as usual in RTL, it's more about about an edge of container where it sticks